subsys/llext/llext_link.c

```diff
@@ -393,64 +595,25 @@ int llext_link(struct llext_loader *ldr, struct llext *ext, const struct llext_l
 
 			op_loc = sect_base + rel.r_offset;
 
-			if (ELF_R_SYM(rel.r_info) == 0) {
-				/* no symbol ex: R_ARM_V4BX relocation, R_ARM_RELATIVE  */
-				link_addr = 0;
-			} else if (sym.st_shndx == SHN_UNDEF) {
-				/* If symbol is undefined, then we need to look it up */
-				link_addr = (uintptr_t)llext_find_sym(NULL,
-					SYM_NAME_OR_SLID(name, sym.st_value));
-
-				if (link_addr == 0) {
-					/* Try loaded tables */
-					struct llext *dep;
-
-					link_addr = (uintptr_t)llext_find_extension_sym(name, &dep);
-					if (link_addr) {
-						llext_dependency_add(ext, dep);
-					}
-				}
-
-				if (link_addr == 0) {
-					LOG_ERR("Undefined symbol with no entry in "
-						"symbol table %s, offset %zd, link section %d",
-						name, (size_t)rel.r_offset, shdr->sh_link);
-					return -ENODATA;
-				}
+			ret = llext_lookup_symbol(&link_addr, &rel, &sym, ext, ldr, name, shdr, i,
+						  j);
```

Comment:
would it fit within 100 characters if you join the two lines? You need to check the error code

Reply:
Added the error check and reformatted the line